src/gui/qgsfieldexpressionwidget.h

```diff
@@ -75,6 +76,20 @@ class GUI_EXPORT QgsFieldExpressionWidget : public QWidget
     //! setFilters allows fitering according to the type of field
     void setFilters( QgsFieldProxyModel::Filters filters );
 
+    /**
+     * Sets whether an optional empty field ("not set") option is shown in the combo box.
+     * \see allowEmptyFieldName()
+     * \since QGIS 3.0
```

Comment:
since 3.8

Reply:
You mean 3.6 ? Made the modifications (with 3.6)